src/main/scala/viper/gobra/frontend/info/base/BuiltInMemberTag.scala

```diff
@@ -0,0 +1,305 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2020 ETH Zurich.
+
+package viper.gobra.frontend.info.base
+
+import org.bitbucket.inkytonik.kiama.==>
+import org.bitbucket.inkytonik.kiama.util.Messaging.{Messages, error, noMessages}
+import viper.gobra.ast.frontend.PExpression
+import viper.gobra.frontend.Config
+import viper.gobra.frontend.info.base.Type.{AssertionT, AbstractType, ChannelModus, ChannelT, FunctionT, IntT, PredT, Type, VoidType}
+import viper.gobra.frontend.info.implementation.typing.ghost.separation.GhostType
+import viper.gobra.util.TypeBounds.UnboundedInteger
+import viper.gobra.util.Violation
+
+
+/**
+  * Module to add built-in functions, methods, fpredicates, and mpredicates to Gobra.
+  * Two steps have to be performed for adding a new built-in member:
+  * (1) add a tag representing that built-in member,
+  * (2) add it to `builtInMembers()`, and
+  * (3) add an additional case to BuiltInMembersImpl that maps that built-in member (the tag and its specific use) to
+  * a generated member in the internal representation which is then encoded.
+  * The desugarer automatically resolves function and methods calls to as well as instances of these built-in members.
+  * Thus, no changes in the desugarer should be necessary.
+  */
+object BuiltInMemberTag {
+  sealed trait BuiltInMemberTag {
+    /** identifier of this member as it appears in the program text */
+    def identifier: String
+    /** debug name for printing */
+    def name: String
+    /** flag whether the entire member is ghost */
+    def ghost: Boolean
+
+    /** abstract type for this tag */
+    def typ(config: Config): AbstractType
+
+    /** ghost typing for arguments */
+    def argGhostTyping(args: Vector[Type])(config: Config): GhostType
+
+    /** ghost typing for return values */
+    def returnGhostTyping(args: Vector[Type])(config: Config): GhostType
+  }
+
+  sealed trait ActualBuiltInMember extends BuiltInMemberTag
+
+  sealed trait GhostBuiltInMember extends BuiltInMemberTag {
+    override def ghost: Boolean = true
+
+    override def argGhostTyping(args: Vector[Type])(config: Config): GhostType =
+      if (typ(config).typing.isDefinedAt(args)) {
+        ghostArgs(typ(config).typing(args).args.length)
+      } else {
+        Violation.violation(s"argGhostTyping not defined for $name")
+      }
+
+    def returnGhostTyping(args: Vector[Type])(config: Config): GhostType =
+      if (typ(config).typing.isDefinedAt(args)) {
+        typ(config).typing(args).result match {
+          case VoidType => ghostArgs(0)
+          case _ => ghostArgs(1) // multi return parameters are not used yet by any built-in member
```

Comment:
I was hoping for the same as well. However this does not seem to work for the following reason:
Assume you want to type a function/method call with 5 arguments and arg typing returns `GhostType.isGhost` i.e. a singleton vector containing `true`. `generalGhostAssignableTo` checks whether this assignment is possible but trying to assign 5 arguments (right) to a single assignee (left) results in an `AssignMode.Error`.
This issue does not seem to occur for user-defined functions and methods because `calleeArgGhostTyping` iterates over all parameters, calls `isParamGhost` and inserts a corresponding flag into GhostType for each parameter.

Reply:
Could you then change the comment to something like "As a default, we pick a single ghost result. Other cases have to overwrite this  function."